# AICTE Approved Institutions Data Processing & Analysis

### Data Source: AICTE Website
### Relevance: 2012-2019

In [1]:
# importing necessary libraries
import pandas as pd
from impyute.imputation.cs import mice

In [2]:
df = pd.read_csv('AICTE_data_Sheet1.csv') #Loading data

In [3]:
df.head()

Year  Total Institutions  New Institutions  Closed Institutions  \
0  2012-13               10272               321                   73   
1  2013-14               10300               157                  108   
2  2014-15               10334               244                   77   
3  2015-16               10329               174                  123   
4  2016-17               10365               293                  155   

   Total Intake  Girl's Enrolment  Boy's Enrolment  Faculty  #Students Passed  \
0       3463402            726105          1431295   392270         1380789.0   
1       3723719            571138          1618179   533352         1504802.0   
2       3961670            554665          1587283   682079         1585743.0   
3       3836181            553264          1532071   698554         1636939.0   
4       3702582            529928          1424844   692191         1709053.0   

   Placement  
0     560640  
1     613106  
2     673837  
3     701031  
4     722494

In [4]:
df.dtypes

Year                    object
Total Institutions       int64
New Institutions         int64
Closed Institutions      int64
Total Intake             int64
Girl's Enrolment         int64
Boy's Enrolment          int64
Faculty                  int64
#Students Passed       float64
Placement                int64
dtype: object

### Handling Missing Values

In [5]:
df.isnull().sum()

Year                   0
Total Institutions     0
New Institutions       0
Closed Institutions    0
Total Intake           0
Girl's Enrolment       0
Boy's Enrolment        0
Faculty                0
#Students Passed       1
Placement              0
dtype: int64

In [6]:
# Filling missing values using MICE algorithm
temp = df.drop(columns=["Year"]) #Excluding Year Column
cols = temp.columns
imputed = pd.DataFrame(mice(temp.values))
imputed.columns = cols
df[cols] = imputed[cols]

In [7]:
df =  df.round() #Rounding off the imputed values

In [8]:
df.describe()

Total Institutions  New Institutions  Closed Institutions  \
count            7.000000          7.000000             7.000000   
mean         10346.571429        294.714286            99.142857   
std             54.313464        130.342772            34.411377   
min          10272.000000        157.000000            53.000000   
25%          10314.500000        209.000000            75.000000   
50%          10334.000000        293.000000           105.000000   
75%          10382.500000        323.500000           115.500000   
max          10426.000000        548.000000           155.000000   

       Total Intake  Girl's Enrolment  Boy's Enrolment        Faculty  \
count  7.000000e+00          7.000000     7.000000e+00       7.000000   
mean   3.661824e+06     568144.428571     1.468611e+06  612032.857143   
std    2.039223e+05      72234.813857     1.130470e+05  116437.779537   
min    3.392834e+06     517115.000000     1.316060e+06  392270.000000   
25%    3.507890e+06     527362.000000     1.397694e+06  560423.000000   
50%    3.702582e+06     553264.000000     1.431295e+06  682079.000000   
75%    3.779950e+06     562901.500000     1.559677e+06  695240.500000   
max    3.961670e+06     726105.000000     1.618179e+06  698554.000000   

       #Students Passed      Placement  
count      7.000000e+00       7.000000  
mean       1.639166e+06  678216.714286  
std        2.400688e+05   69375.440356  
min        1.380789e+06  560640.000000  
25%        1.515942e+06  643471.500000  
50%        1.585743e+06  701031.000000  
75%        1.672996e+06  718534.000000  
max        2.129757e+06  761835.000000

In [9]:
df

Year  Total Institutions  New Institutions  Closed Institutions  \
0  2012-13             10272.0             321.0                 73.0   
1  2013-14             10300.0             157.0                108.0   
2  2014-15             10334.0             244.0                 77.0   
3  2015-16             10329.0             174.0                123.0   
4  2016-17             10365.0             293.0                155.0   
5  2017-18             10400.0             326.0                105.0   
6  2018-19             10426.0             548.0                 53.0   

   Total Intake  Girl's Enrolment  Boy's Enrolment   Faculty  \
0     3463402.0          726105.0        1431295.0  392270.0   
1     3723719.0          571138.0        1618179.0  533352.0   
2     3961670.0          554665.0        1587283.0  682079.0   
3     3836181.0          553264.0        1532071.0  698554.0   
4     3702582.0          529928.0        1424844.0  692191.0   
5     3552377.0          524796.0        1370544.0  698290.0   
6     3392834.0          517115.0        1316060.0  587494.0   

   #Students Passed  Placement  
0         1380789.0   560640.0  
1         1504802.0   613106.0  
2         1585743.0   673837.0  
3         1636939.0   701031.0  
4         1709053.0   722494.0  
5         1527081.0   714574.0  
6         2129757.0   761835.0

### Finding Correlation

In [10]:
df.corr()

Total Institutions  New Institutions  \
Total Institutions             1.000000          0.682032   
New Institutions               0.682032          1.000000   
Closed Institutions           -0.082626         -0.567179   
Total Intake                  -0.321741         -0.762601   
Girl's Enrolment              -0.788505         -0.141542   
Boy's Enrolment               -0.677334         -0.869700   
Faculty                        0.574206         -0.138281   
#Students Passed               0.773350          0.720775   
Placement                      0.922434          0.471566   

                     Closed Institutions  Total Intake  Girl's Enrolment  \
Total Institutions             -0.082626     -0.321741         -0.788505   
New Institutions               -0.567179     -0.762601         -0.141542   
Closed Institutions             1.000000      0.410971         -0.281173   
Total Intake                    0.410971      1.000000         -0.236256   
Girl's Enrolment               -0.281173     -0.236256          1.000000   
Boy's Enrolment                 0.223069      0.819089          0.116696   
Faculty                         0.485692      0.528210         -0.846749   
#Students Passed               -0.256133     -0.307436         -0.608460   
Placement                       0.137263     -0.031840         -0.884098   

                     Boy's Enrolment   Faculty  #Students Passed  Placement  
Total Institutions         -0.677334  0.574206          0.773350   0.922434  
New Institutions           -0.869700 -0.138281          0.720775   0.471566  
Closed Institutions         0.223069  0.485692         -0.256133   0.137263  
Total Intake                0.819089  0.528210         -0.307436  -0.031840  
Girl's Enrolment            0.116696 -0.846749         -0.608460  -0.884098  
Boy's Enrolment             1.000000  0.025474         -0.514898  -0.482027  
Faculty                     0.025474  1.000000          0.282104   0.768813  
#Students Passed           -0.514898  0.282104          1.000000   0.794087  
Placement                  -0.482027  0.768813          0.794087   1.000000

> Placements are highly correlated to the total number of institution (0.922434)

> Placements are highly correlated to the total number of faculties (0.768813)



In [11]:
df

Year  Total Institutions  New Institutions  Closed Institutions  \
0  2012-13             10272.0             321.0                 73.0   
1  2013-14             10300.0             157.0                108.0   
2  2014-15             10334.0             244.0                 77.0   
3  2015-16             10329.0             174.0                123.0   
4  2016-17             10365.0             293.0                155.0   
5  2017-18             10400.0             326.0                105.0   
6  2018-19             10426.0             548.0                 53.0   

   Total Intake  Girl's Enrolment  Boy's Enrolment   Faculty  \
0     3463402.0          726105.0        1431295.0  392270.0   
1     3723719.0          571138.0        1618179.0  533352.0   
2     3961670.0          554665.0        1587283.0  682079.0   
3     3836181.0          553264.0        1532071.0  698554.0   
4     3702582.0          529928.0        1424844.0  692191.0   
5     3552377.0          524796.0        1370544.0  698290.0   
6     3392834.0          517115.0        1316060.0  587494.0   

   #Students Passed  Placement  
0         1380789.0   560640.0  
1         1504802.0   613106.0  
2         1585743.0   673837.0  
3         1636939.0   701031.0  
4         1709053.0   722494.0  
5         1527081.0   714574.0  
6         2129757.0   761835.0

### Feature Extraction

In [12]:
df["Not_placed"] = df["#Students Passed"] - df["Placement"]

In [13]:
df["placement_ratio"] = df["Placement"]/df["#Students Passed"]

In [14]:
df

Year  Total Institutions  New Institutions  Closed Institutions  \
0  2012-13             10272.0             321.0                 73.0   
1  2013-14             10300.0             157.0                108.0   
2  2014-15             10334.0             244.0                 77.0   
3  2015-16             10329.0             174.0                123.0   
4  2016-17             10365.0             293.0                155.0   
5  2017-18             10400.0             326.0                105.0   
6  2018-19             10426.0             548.0                 53.0   

   Total Intake  Girl's Enrolment  Boy's Enrolment   Faculty  \
0     3463402.0          726105.0        1431295.0  392270.0   
1     3723719.0          571138.0        1618179.0  533352.0   
2     3961670.0          554665.0        1587283.0  682079.0   
3     3836181.0          553264.0        1532071.0  698554.0   
4     3702582.0          529928.0        1424844.0  692191.0   
5     3552377.0          524796.0        1370544.0  698290.0   
6     3392834.0          517115.0        1316060.0  587494.0   

   #Students Passed  Placement  Not_placed  placement_ratio  
0         1380789.0   560640.0    820149.0         0.406029  
1         1504802.0   613106.0    891696.0         0.407433  
2         1585743.0   673837.0    911906.0         0.424935  
3         1636939.0   701031.0    935908.0         0.428257  
4         1709053.0   722494.0    986559.0         0.422745  
5         1527081.0   714574.0    812507.0         0.467935  
6         2129757.0   761835.0   1367922.0         0.357710

In [15]:
df.to_csv('preprocessed_aicte.csv', index=False)